edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/TwinDiffE2ETest.cs

```diff
@@ -377,6 +377,26 @@ static async Task<(DeviceClient, string)> InitializeDeviceClient(RegistryManager
                 }
             }
 
+            for (int i = 0; i < 5; i++)
+            {
+                try
+                {
+                    var device = await rm.GetDeviceAsync(deviceName);
+                    if (device.ConnectionState != DeviceConnectionState.Connected)
+                        throw new Exception("Device not connected to cloud");
+                    break;
+                }
+                catch (Exception)
+                {
+                    if (i == 4)
+                    {
+                        throw;
+                    }
+
+                    await Task.Delay(TimeSpan.FromSeconds(5));
```

Comment:
Is this for the same reason as the delay of 5s that Vadim commented on? Just to not call iothub a lot?

Reply:
yes, it checks that device was connected to cloud